src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -621,7 +621,7 @@ class Invariance : public StackObj {
 // Returns true if the predicate of iff is in "scale*iv + offset u< load_range(ptr)" format
 // Note: this function is particularly designed for loop predication. We require load_range
 //       and offset to be loop invariant computed on the fly by "invar"
-bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invariance& invar) const {
+bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invariance& invar, ProjNode *predicate_proj) const {
```

Comment:
`predicate_proj` is used only for assert. Use `DEBUG_ONLY(, ProjNode *predicate_proj)`.
Unless you want to do the check in PRODUCT.